src/app/components/authDialog/register/register.component.html

```diff
@@ -1,17 +1,48 @@
-<div mat-dialog-content>
-  <mat-form-field>
-    <input matInput [(ngModel)]="username" placeholder="Username">
-  </mat-form-field>
-  <mat-form-field>
-    <input matInput type="email" [(ngModel)]="email" placeholder="Email">
-  </mat-form-field>
-  <mat-form-field>
-    <input matInput type="password" [(ngModel)]="password" placeholder="Password">
-  </mat-form-field>
-  <mat-form-field>
-    <input matInput type="password" [(ngModel)]="repeatedPassword" placeholder="Repeat Password">
-  </mat-form-field>
-</div>
-<div mat-dialog-actions>
-  <rm-loading-button [disabled]="!validate()" [loading]="loading" (push)="register()">Register</rm-loading-button>
-</div>
+<form [formGroup]="myForm">
```

Comment:
After quick look at https://material.angular.io/components/form-field/examples it seems there is a simpler way to add validation (aka no need for form tag)

Also I think all 
<mat-error *ngIf="myForm.get('emailValidate').errors
+        && myForm.get('emailValidate').dirty
+        && myForm.get('emailValidate').errors.errorUrl"
+      >

can be checked easier emailValidate.invalid (I would rename this to emailValidation or emailValidator)